Examples/UIExplorer/UIExplorerUnitTests/RCTImageLoaderHelpers.m

```diff
@@ -47,9 +47,9 @@ - (BOOL)canLoadImageURL:(NSURL *)requestURL
   return _canLoadImageURLHandler(requestURL);
 }
 
-- (RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL size:(CGSize)size scale:(CGFloat)scale resizeMode:(RCTResizeMode)resizeMode progressHandler:(RCTImageLoaderProgressBlock)progressHandler completionHandler:(RCTImageLoaderCompletionBlock)completionHandler
+- (RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL bundlePath:(NSString *)bundlePath size:(CGSize)size scale:(CGFloat)scale resizeMode:(RCTResizeMode)resizeMode progressHandler:(RCTImageLoaderProgressBlock)progressHandler completionHandler:(RCTImageLoaderCompletionBlock)completionHandler
```

Comment:
While I could move the bundle finding logic up to the point where the NSURL is created, we would then need to break apart the URL _[1]_ at this point in order to load it from the relevant bundle.

This would also involve a much larger functional change as now all local images would be loaded differently whereas with the current code they are only loaded differently when specifying a custom bundle path _[2]_ _[3]_.

It also adds another point of failure; the ```RCTBundlePathForURL``` is actually fairly complex and it assumes the asset is in the mainBundle; so in order to safely (assuming that method uses the correct approach) break apart the URL we would likely need to loop through all of the bundles anyway in order to recreate the above method with the ability to specify a bundle.

But seriously, if you do know of a way to load an asset from a bundle with a given URL or can think of a nicer way to do this (knowing the above), then let me know because I'd be happy to reduce the amount of code needed for this PR.

(And sorry for the massive reply)

----------------------------------------------------------------
_1._ While we might be able to load an image from another bundle using ```[UIImage initWithContentsOfFile:]``` it unfortunately doesn't seem to work for assets which have been included in an '.xcassets' file (those are built in to a .car file and cannot be accessed directly as far as I can tell). Which means we have to use the ```[UIImage imageNamed:inBundle:compatibleWithTraitCollection:]``` method.

_2._ ```[UIImage imageName:]``` just defaults to the mainBundle.

_3._ If we ignored the .xcassets case there is still an issue where the caching behaviour between [UIImage imageNamed:inBundle:compatibleWithTraitCollection:] and [UIImage imageWithContentsOfFile:] actually differ. ```imageWithContentsOfFile:``` doesn't load the image in to the system image cache which could affect performance when loading lots of reusable images rapidly [https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIImage_Class/index.html#//apple_ref/doc/uid/TP40006890-CH3-SW10](url)